charts/injector/templates/webhook.yaml

```diff
@@ -0,0 +1,36 @@
+
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: MutatingWebhookConfiguration
+metadata:
+  name: {{ include "charts.fullname" . }}
+  labels:
+    {{- include "charts.labels" . | nindent 4 }}
+webhooks:
+  - name: {{ include "charts.fullname" . }}.{{ .Release.Namespace }}.svc
+{{- with .Values.webhook }}
+    matchPolicy: Equivalent
+    reinvocationPolicy: Never
+    failurePolicy: Fail
+    timeoutSeconds: 30
+    namespaceSelector:
+      {{- toYaml .namespaceSelector | nindent 6 }}
+    objectSelector:
+      {{- toYaml .objectSelector | nindent 6 }}
+    clientConfig:
+      # kubectl config view --raw --minify --flatten -o jsonpath='{.clusters[].cluster.certificate-authority-data}'
+      # caBundle is the CA cert that sign the webhook's serving cert
+      caBundle: {{ required "webhook.caBundle must be set" .caBundle }}
+{{- end }}
+      service:
+        name: {{ include "charts.fullname" . }}
+        namespace: {{ .Release.Namespace }}
+        path: "/mutate"
+    rules:
+      - operations: ["CREATE"]
+        apiGroups: ["apps"]
+        apiVersions: ["v1"]
+        resources: ["deployments"]
+      - operations: ["CREATE"]
+        apiGroups: ["apps"]
+        apiVersions: ["v1"]
+        resources: ["statefulsets"]
```

Comment:
maybe we should also add this section to optional and decided by `values.yaml` file.